tests/plugins/test_deepspeed_plugin.py

```diff
@@ -739,6 +742,51 @@ def test_deepspeed_multigpu_test(tmpdir, deepspeed_config):
     trainer.test(model)
 
 
+@RunIf(min_gpus=2, deepspeed=True, special=True)
+def test_deepspeed_multigpu_partial_partition_parameters(tmpdir):
+    """Test to ensure that a module that defines a layer inside the ``__init__`` and ``configure_sharded_model``
+    correctly converts all parameters to float16 when ``precision=16`` and runs successfully."""
+
+    class TestModel(ModelParallelBoringModel):
+        def __init__(self):
+            super().__init__()
+            self.layer_2 = torch.nn.Linear(32, 32)
+
+        def forward(self, x):
+            x = self.layer_2(x)
+            return self.layer(x)
```

Comment:
maybe define the configure_sharded layer initialization explicitly here instead of relying on super class, especially since there are no assertions in this test that would otherwise fail when the super class changes for some reason